.github/workflows/build_packages.yml

```diff
@@ -57,26 +67,50 @@ jobs:
     permissions:
       contents: read
 
+    outputs:
+      tar_cache: ${{ steps.cudaq_image.outputs.tar_cache }}
+      tar_archive: ${{ steps.cudaq_image.outputs.tar_archive }}
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
 
       - name: Restore release build
         uses: actions/cache/restore@v3
         with:
-          path: /tmp/cudaqdev.tar
+          path: ${{ needs.release_build.outputs.cudaqdev_tarfile }}
           key: ${{ needs.release_build.outputs.cudaqdev_cache }}
           fail-on-cache-miss: true
 
       - name: Build cuda-quantum image
+        id: cudaq_image
         run: |
-          docker load --input /tmp/cudaqdev.tar
-          docker build -t cuda-quantum:latest -f docker/release/cudaq.Dockerfile . \
+          repo_owner=${{ github.repository_owner }}
+          image_name=${{ vars.registry || 'ghcr.io' }}/${repo_owner,,}/cuda-quantum
+          tag_prefix=${{ (github.event.pull_request && 'pr-') || '' }}
+          tag_name=`echo ${{ github.ref_name }} | cut -d '/' -f 1`
+          image_tag=${tag_prefix}${tag_name}
+          tar_archive=/tmp/cuda-quantum.tar
+
+          docker load --input ${{ needs.release_build.outputs.cudaqdev_tarfile }}
+          docker build -t $image_name:$image_tag -f docker/release/cudaq.Dockerfile . \
             --build-arg dev_image=cuda-quantum-dev --build-arg dev_tag=local
+          docker save $image_name:$image_tag > $tar_archive
+
+          echo "image_name=$image_name" >> $GITHUB_OUTPUT
+          echo "image_tag=$image_tag" >> $GITHUB_OUTPUT
+          echo "tar_archive=$tar_archive" >> $GITHUB_OUTPUT
+          echo "tar_cache=tar-cuda-quantum-${{ github.sha }}" >> $GITHUB_OUTPUT
+
+      - name: Cache cuda-quantum image
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ steps.cudaq_image.outputs.tar_archive }}
+          key: ${{ steps.cudaq_image.outputs.tar_cache }}
 
       - name: Validate cuda-quantum image
         run: |
-          docker run --rm -dit --name cuda-quantum cuda-quantum:latest
+          docker run --rm -dit --name cuda-quantum ${{ steps.cudaq_image.outputs.image_name }}:${{ steps.cudaq_image.outputs.image_tag }}
           docker cp scripts/validate_container.sh cuda-quantum:"/home/cudaq/validate_container.sh"
           docker exec -e TERM=xterm cuda-quantum bash validate_container.sh > /tmp/validation.out
```

Comment:
I think we need to reconsider squirrelling away output into various temporary files exclusively. It's difficult for people not familiar with the details to figure out what happened when things break.

Some possibilities:
- use `tee` to both capture the output and let it be printed in the stdout stream.
- postprocess such that the captured output, if not empty, is concatenated to stdout.

Reply:
I am not sure I follow what you suggest here; are you asking if we can show the output of validate_container.sh in the log with the rest of what the step does rather than only in the step summary?